packages/@react-aria/overlays/test/useOverlay.test.js

```diff
@@ -11,57 +11,101 @@
  */
 
 import {fireEvent, render} from '@testing-library/react';
+import {installMouseEvent, installPointerEvent} from '@react-spectrum/test-utils';
+import {mergeProps} from '@react-aria/utils';
 import React, {useRef} from 'react';
 import {useOverlay} from '../';
 
 function Example(props) {
   let ref = useRef();
-  let {overlayProps} = useOverlay(props, ref);
-  return <div ref={ref} {...overlayProps} data-testid={props['data-testid'] || 'test'}>{props.children}</div>;
+  let {overlayProps, underlayProps} = useOverlay(props, ref);
+  return (
+    <div {...mergeProps(underlayProps, props.underlayProps || {})}>
+      <div ref={ref} {...overlayProps} data-testid={props['data-testid'] || 'test'}>
+        {props.children}
+      </div>
+    </div>
+  );
 }
 
 describe('useOverlay', function () {
-  it('should not focus the overlay if a child is focused', function () {
-    let res = render(
-      <Example isOpen>
-        <input autoFocus data-testid="input" />
-      </Example>
-    );
-
-    let input = res.getByTestId('input');
-    expect(document.activeElement).toBe(input);
-  });
+  describe.each`
```

Comment:
Shouldn't we do onTouch and onClick too?

Reply:
I don't understand what you're asking, this tests mouse/pointer/touch